### PR TITLE
ci: enable PyLint for pull-requests

### PR DESCRIPTION
**Pull request by praiskup**

### CODE REVIEW

**Comment by praiskup** [comment]
@zmiklank do you think this change makes sense for this project?

File: .github/workflows/python-diff-lint.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - id: VCS_Diff_Lint
+        name: VCS Diff Lint
+        uses: fedora-copr/vcs-diff-lint-action@v1
+
+      - name: Upload artifact with detected defects in SARIF format
+        uses: actions/upload-artifact@v3
+        with:
+          name: VCS Diff Lint SARIF
+          path: ${{ steps.VCS_Diff_Lint.outputs.sarif }}
+        if: ${{ always() }}
```
Comment (zmiklank) [inline .github/workflows/python-diff-lint.yml:30]: Is this needed? I think that even without the condition the jobs will run sequentially. [1]


[1] https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-not-requiring-successful-dependent-jobs

**Comment by zmiklank** [comment]
> @zmiklank do you think this change makes sense for this project?

@praiskup Yes, I think pylint for every pull request is great thing. I personally sometimes forget to run linters before creating PR, so this would be a big help also for me.

**Review by zmiklank — COMMENTED** [review]
I, however, can not really do a proper review, as I am not sure what pylint settings are needed in the pylintrc that you included.

File: .github/workflows/python-diff-lint.yml (hunk 1 above)
Comment (praiskup) [reply .github/workflows/python-diff-lint.yml:30]: Yes, per instructions in README.md: https://github.com/fedora-copr/vcs-diff-lint-action

The `vcs-diff-lint-action` can fail (some new Python errors created in PR => FAIL), and you still want to upload the SARIF.

Comment (praiskup) [reply .github/workflows/python-diff-lint.yml:30]: See this example run: https://github.com/devexp-db/distgen/actions/runs/4611524352/jobs/8151315618

**Comment by praiskup** [comment]
> I, however, can not really do a proper review, as I am not sure what pylint settings are needed in the pylintrc that you included.

That is just some default that we use in other projects;  it can be tweaked later.  If you want me to fix
something right now, please tell.

File: .github/workflows/python-diff-lint.yml (hunk 1 above)
Comment (zmiklank) [reply .github/workflows/python-diff-lint.yml:30]: Right. Thanks for explanation.

**Comment by zmiklank** [comment]
> > I, however, can not really do a proper review, as I am not sure what pylint settings are needed in the pylintrc that you included.
> 
> That is just some default that we use in other projects; it can be tweaked later. If you want me to fix something right now, please tell.

I am OK with this default.

**Comment by praiskup** [comment]
> I am OK with this default.

So the change is blessed!  Thank you for looking.

**Pull request merged**